viewing_party/party.py

```diff
@@ -1,23 +1,285 @@
+#     - Each friend in `"friends"` has a watched list. Each movie in the watched list has a `"host"`, which is a string that says what streaming service it's hosted on
+# - Determine a list of recommended movies. A movie should be added to this list if and only if:
+#   - The user has not watched it
+#   - At least one of the user's friends has watched
+#   - The `"host"` of the movie is a service that is in the user's `"subscriptions"`
+# - Return the list of recommended movies
 # -----------------------------------------
+def get_available_recs(user_data):
+    
+    subs = user_data["subscriptions"]
+    avail_recs = []
+    no_recs = []
+    if get_friends_unique_watched == []:
+        return no_recs
+    else:
+        for streaming_services in subs:
+            for unique_movie in get_friends_unique_watched(user_data):
+                if streaming_services == unique_movie['host']:
+                    avail_recs.append(unique_movie)
+        return avail_recs
+
+    
+
+
 
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
+# 1. There are four tests about a `get_new_rec_by_genre` function
+
+# Create a function named `get_new_rec_by_genre`
+
+# - takes one parameter: `user_data`
+# - Consider the user's most frequently watched genre. Then, determine a list of recommended movies. A movie should be added to this list if and only if:
+#   - The user has not watched it
+#   - At least one of the user's friends has watched
+#   - The `"genre"` of the movie is the same as the user's most frequent genre
+# - Return the list of recommended movies
+
 # -----------------------------------------
 
+
+def get_new_rec_by_genre(user_data):
+    # find users most watched genre
+    # add movies to watch list if user hasnt watched it , a friend has watched it, genre matches most watched
+    # return list of movies 
+    rec_by_genre = []
+    no_recs= []
+    genre = get_most_watched_genre(user_data)
+    movie_list = get_friends_unique_watched(user_data)
+    if movie_list == []:
+        return no_recs 
+    else: 
+        for movie in movie_list:
+            if genre == movie['genre']:
+                rec_by_genre.append(movie)
+            return rec_by_genre
+
+# 2. There are also two tests about a `get_rec_from_favorites` function
+
+# Create a function named `get_rec_from_favorites`
+
+# - takes one parameter: `user_data`
+#   - `user_data` will have a field `"favorites"`. The value of `"favorites"` is a list of movie dictionaries
+#     - This represents the user's favorite movies
+# - Then, determine a list of recommended movies. A movie should be added to this list if and only if:
+#   - The movie is in the user's `"favorites"`
+#   - None of the user's friends have watched it
+# - Return the list of recommended movies
+def get_rec_from_favorites(user_data):
+    favorite_recs = []
+    user_unique = get_unique_watched(user_data)

```

Comment:
You could consider renaming `user_unique` to `unique_user_movies` which is a little more descriptive and implies that this variable is a data structure with more than 1 movie in it.